File: first assignment/python/vertex.py
from typing import Dict

class Vertex:
    def __init__(self, name: str, key: int, parent = None) -> None:
        self.name = name
        self.key = key
        self.parent = parent
        self.vertices_adjacent: Dict[str, Vertex] = {}
        self.edges_incident = {}
        self.visited = False

    def __eq__(self, other: object) -> bool:
        return isinstance(other, self.__class__) and self.name == other.name

    def __lt__(self, other: object) -> bool:
        return self.key < other.key

    def add_incident_edge(self, edge) -> None:
        if not (edge.u.name, edge.v.name) in self.edges_incident:
            self.edges_incident[(edge.u.name, edge.v.name)] = edge

    def remove_incident_edge(self, edge) -> None:
        if (edge.u.name, edge.v.name) in self.edges_incident:
            del self.edges_incident[(edge.u.name, edge.v.name)]

    def add_adjacent_vertices(self, vertex) -> None:
        if not vertex.name in self.vertices_adjacent:
            self.vertices_adjacent[vertex.name] = vertex

    def remove_adjacent_vertices(self, vertex) -> None:
        if vertex.name in self.vertices_adjacent:
            del self.vertices_adjacent[vertex.name]

File: second assignment/c#/SecondAssignment/SecondAssignment/Vertex.cs
namespace SecondAssignment;

public class Vertex : IEquatable<Vertex>
{
    public string Name { get; set; }
    public double X { get; init; }
    public double Y { get; init; }

    public bool Visited { get; set; }
    public double Key { get; set; }
    public Vertex? Parent { get; set; }
    public Dictionary<string, Vertex> VerticesAdjacent { get; set; }

    public Vertex(string name, double x, double y, Type type)
    {
        Name = name;
        X = x;
        Y = y;
        VerticesAdjacent = new();
        Key = 0;

        if (type == Type.GEO)
        {
            X = ConvertGeoCoordinate(X);
            Y = ConvertGeoCoordinate(Y);
        }
    }

    private static double ConvertGeoCoordinate(double x)
    {
        const double PI = 3.141592;
        double deg = Math.Truncate(x);
        double min = x - deg;
        double rad = PI * (deg + 5.0 * min / 3.0) / 180.0;
        return rad;
    }

    public bool IsVisited() => Visited;
    public void SetVisited(bool value) => Visited = value;

    public void AddAdjacentVertices(Vertex v)
    {
        if (!VerticesAdjacent.ContainsKey(v.Name))
            VerticesAdjacent.Add(v.Name, v);
    }

    public void RemoveAdjacentVertices(Vertex v)
    {
        if (VerticesAdjacent.ContainsKey(v.Name))
            VerticesAdjacent.Remove(v.Name);
    }

    public void ClearStatus()
    {
        Visited = false;
        Key = 0;
    }

    public bool Equals(Vertex? other) => other != null && Name.Equals(other.Name);
}